src/server.rs

```diff
@@ -491,6 +492,20 @@ impl<State: Sync + Send + 'static, InnerState: Sync + Send + 'static> Endpoint<S
     }
 }
 
+async fn peek_disconnect(stream: async_std::net::TcpStream) -> http_types::Result<()> {
+    let mut buf: Vec<u8> = vec![0; 1];
+    loop {
+        match stream.peek(&mut buf).await {
```

Comment:
How often does this happen? Should this be on a timer?

Reply:
I initially thought so, but debugging indicates it only loops once (as `Ok(0)`) on my machine. I'm not entirely sure why, and this might be platform-dependent